sdk/servicebus/Microsoft.Azure.ServiceBus/src/Message.cs

```diff
@@ -295,18 +295,26 @@ public override string ToString()
         /// <summary>Clones a message, so that it is possible to send a clone of an already received
         /// message as a new message. The system properties of original message
         /// are not copied.</summary>
+        /// <remarks>The <see cref="UserProperties"/> is a new dictionary containing references to the original entries.</remarks>
         /// <returns>A cloned <see cref="Message" />.</returns>
         public Message Clone()
         {
             var clone = (Message)this.MemberwiseClone();
-            clone.SystemProperties = new SystemPropertiesCollection();
 
             if (this.Body != null)
             {
                 var clonedBody = new byte[this.Body.Length];
                 Array.Copy(this.Body, clonedBody, this.Body.Length);
                 clone.Body = clonedBody;
             }
+
+            clone.SystemProperties = new SystemPropertiesCollection();
+            clone.UserProperties = new Dictionary<string, object>(this.UserProperties.Count);
```

Comment:
Should the property values themselves also be cloned?

Reply:
This is a difficult question, because it would not be easy to perfectly clone the values. Since the value in the key-value pair of dictionary is an object, how would you clone it? Lets say the object is a `List<object>`. Even if I clone the list itself, I don't know a way to clone individual objects themselves.
Given that there's no perfect implementation, current approach has been to provide a reasonable mid-point. 
Maybe allowing properties to be added/removed from the cloned message without affecting the original message could be a fair mid-point IMO. 
The main purpose of exposing `Clone()` method was to allow sending a received message to a different queue without the extra headers that the broker added on the way back.